### ISSUE TITLE
fix: Restrict top-level workflow token permissions (Token-Permissions)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#254]
## Overview

OpenSSF Scorecard flags `Token-Permissions` as 0/10 because `.github/workflows/goreleaser.yml` declares `permissions: contents: write` at the workflow level. This grants write access to all jobs in the workflow, including jobs that only need read access.

## Problem

Scorecard requires that every workflow either sets `permissions: read-all` / `permissions: {}` at the top level and then grants only the minimum write permissions at the job level. The current top-level `contents: write` in `goreleaser.yml` violates this principle.

```yaml
# current — flagged by Scorecard
permissions:
  contents: write
```

## Proposed Fix

Set `permissions: read-all` at the workflow top level (matching the pattern already used in `scorecard.yml`). Each job already declares its own job-level permissions, so no job behaviour changes.

```yaml
# after
permissions: read-all
```

Job-level permissions already in place (no changes needed):

| Job | permissions |
|---|---|
| `release` | `contents: write` |
| `update-major-tag` | `contents: write` |
| `push-docker` | `contents: read`, `packages: write` |
| `npm-publish` | `contents: read`, `id-token: write` |

## References

- [Scorecard Token-Permissions check](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions)

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create

None.

### Files to modify

- `.github/workflows/goreleaser.yml` — change top-level `permissions: contents: write` to `permissions: read-all`

### Checklist

- [ ] Change `permissions: contents: write` → `permissions: read-all` at workflow top level in `goreleaser.yml`
- [ ] Verify all jobs already have their own job-level permissions (no job should silently rely on the top-level grant)
- [ ] Confirm release workflow still passes on next tag push
